### PR TITLE
Prefix internal channels used for invoke/handle

### LINKED ISSUE #361: invoke gets handled in both processes if channel is same
I found a bug where calling `ipcRenderer.invoke()` will send a message to both the main and renderer processes if they are both listening.

Here's some code:

```ts
test("ipc with same name", async () => {
  ipcRenderer.on("hello", () => {
    console.log("renderer got hello")
  })
  ipcMain.handle("hello", () => {
    console.log("main got hello")
  })
  await ipcRenderer.invoke("hello")
})
```

The output looks like:

```
    main got hello
    renderer got hello
```

### PR DESCRIPTION
**Pull request by jameskerr**
Closes #361 

When implementing invoke and handle, prefix the channel name so that it will not collide with other user-defined listeners.

I chose the prefix: `__electron_mock_ipc__:`

I've added a test in here too.

Let me know if there's anything more I can do! And thank you for a great library.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "main": "lib/src/index.js",
   "typings": "./lib/src/index.d.ts",
   "scripts": {
+    "prepare": "npm run build",
```
Comment (h3poteto) [inline package.json:8]: Why define this new script? It seems that this script is same as `build`.

File: src/pipe.ts [hunk 2]
```diff
@@ -1,13 +0,0 @@
...
-const pipe = (main: ipcMain, renderer: ipcRenderer) => {
-  main.emitter.on('send-to-renderer', (channel: string, ...args: any) => {
-    setTimeout(() => renderer.emitter.emit('receive-from-main', channel, ...args), 1)
-  })
-  renderer.emitter.on('send-to-main', (channel: string, ...args: any) => {
-    setTimeout(() => main.emitter.emit('receive-from-renderer', channel, ...args), 1)
-  })
-}
-
-export default pipe
```
Comment (h3poteto) [inline src/pipe.ts:13]: Thank you. This file is no longer needed.

File: package.json (hunk 1 above)
Comment (jameskerr) [reply package.json:8]: Yes, the prepare script was needed for me to get my fork working in my app's package.json. When a prepare script is present during an `npm install` (in an app), the dev dependencies get downloaded and the prepare script gets run. 

But if this is blocking you from merging, I can take it out.

**Review by h3poteto — APPROVED** [review]
Thank you!

**Pull request merged**